test/correctness/output_larger_than_two_gigs.cpp

```diff
@@ -11,29 +11,39 @@ void halide_error(void *ctx, const char *msg) {
 using namespace Halide;
 
 int main(int argc, char **argv) {
+    Target t = get_target_from_environment();
+
     Var x, y, z;
     Func identity_uint8;
     identity_uint8(x, y, z) = cast<uint8_t>(42);
-
-    uint8_t c[4096];
-    memset(c, 42, sizeof(c));
-
-    buffer_t buf;
-    memset(&buf, 0, sizeof(buf));
-    buf.host = c;
-    buf.extent[0] = 4096;
-    buf.extent[1] = 4096;
-    buf.extent[2] = 256;
-    buf.stride[0] = 1;
-    buf.stride[1] = 0;
-    buf.stride[2] = 0;
-    buf.elem_size = 1;
-
     identity_uint8.set_error_handler(&halide_error);
 
-    Buffer output_buf(UInt(8), &buf);
-    identity_uint8.realize(output_buf);
-
-    assert(error_occurred);
+    if (t.bits == 64) {
+        Image<uint8_t> output(4096, 4096, 256);
```

Comment:
You can play games with the buffer_t (set some of the strides to zero) to
make it appear to be a huge buffer without actually taking up a lot of
memory. I believe that's what the original code did.

On Fri, Oct 23, 2015 at 11:37 AM, Tyler Denniston <notifications@github.com>
wrote:

> In test/correctness/output_larger_than_two_gigs.cpp
> <https://github.com/halide/Halide/pull/952#discussion_r42899157>:
>
> >      identity_uint8.set_error_handler(&halide_error);
> >
> > -    Buffer output_buf(UInt(8), &buf);
> > -    identity_uint8.realize(output_buf);
> > -
> > -    assert(error_occurred);
> > +    if (t.bits == 64) {
> > +        Image<uint8_t> output(4096, 4096, 256);
>
> I'm not sure how to rewrite these tests without actually allocating an
> image >2GB.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/halide/Halide/pull/952/files#r42899157>.
>